tree/ntuple/v7/test/ntuple_basics.cxx

```diff
@@ -639,3 +639,61 @@ TEST(RNTuple, BareEntry)
    ntuple->LoadEntry(1);
    EXPECT_EQ(2.0, *ntuple->GetModel()->GetDefaultEntry()->Get<float>("pt"));
 }
+
+namespace ROOT::Experimental::Internal {
+struct RFieldCallbackInjector {
+   template <typename FieldT>
+   static void Inject(FieldT &field, ROOT::Experimental::Detail::RFieldBase::ReadCallback_t func)
+   {
+      field.AddReadCallback(func);
+   }
+};
+} // namespace ROOT::Experimental::Internal
+namespace {
+static unsigned gNCallReadCallback = 0;
+}
+
+TEST(RNTuple, ReadCallback)
+{
+   using RFieldCallbackInjector = ROOT::Experimental::Internal::RFieldCallbackInjector;
+
+   FileRaii fileGuard("test_ntuple_readcb.ntuple");
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldI32 = model->MakeField<std::int32_t>("i32", 0);
+      auto fieldKlass = model->MakeField<CustomStruct>("klass");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "f", fileGuard.GetPath());
+      fieldKlass->a = 42.0;
+      fieldKlass->s = "abc";
+      ntuple->Fill();
+      *fieldI32 = 1;
+      fieldKlass->a = 24.0;
+      ntuple->Fill();
+   }
+
+   auto model = RNTupleModel::Create();
+   auto fieldI32 = std::make_unique<RField<std::int32_t>>("i32");
+   auto fieldKlass = std::make_unique<RField<CustomStruct>>("klass");
+   RFieldCallbackInjector::Inject(*fieldI32, [](RFieldValue &value) {
```

Comment:
Why is this syntax better than:

```suggestion
   fieldI32->AddReadCallback( [](RFieldValue &value) {
```

Reply:
`AddReadCallback()` is protected in this context.  In principle, these functions are considered an internal affair for now.

Reply:
Humm .... what is the API that ATLAS would be using (and/or is it not decided yet)?

Reply:
This PR only includes the infrastructure for supporting post-read callbacks.  The user-facing interface is yet to be decided, but should be probably based on ROOT I/O rules if possible.

Reply:
Fair enough.